arch/x86/core/ia32/crt0.S

```diff
@@ -260,7 +261,13 @@ __csSet:
 	call	z_loapic_enable
 #endif
 
-	pushl	%ebx		/* pointer to multiboot info, or NULL */
+	movl	$x86_cpu_boot_arg, %ebp
+	/* Boot type to multiboot, ebx content will help to mitigate */
```

Comment:
Not sure there's much value in this being a runtime detection.  It doesn't hurt anything really, but in general Zephyr always knows the platform is running on and we'll never need to boot the same binary from both grub and EFI.

Reply:
I did not change the existing build, i.e.: both zephyr.strip and zephyr.efi are built. Perhaps that would be the next step, when zefi will be using the same toolchain and will be aware of the CONFIG_*. At this point it will be possible indeed to get rid of the boot_type (unless we make this zefi work first, then I'll rebase this PR against it)